src/Components/Action.vue

```diff
@@ -55,16 +56,13 @@ export default {
 				this.$parent._$el.fadeOut(OC.menuSpeed)
 				this.$parent.$emit('remove')
 				try {
-					$('body').trigger(new $.Event('OCA.Notification.Action', {
-						notification: this.$parent,
-						action: {
-							url: this.link,
-							type: type,
-						},
-					}))
-				// do not do anything but log, the action went fine
-				// only the event bus listener failed, this is not our problem
+					emit('notifications:action', {
+						notification: this.$parent._props,
```

Comment:
I'm not super confident about using `_props`. What is this about?

Reply:
basically all the api data:
https://github.com/nextcloud/notifications/blob/master/docs/ocs-endpoint-v2.md#getting-the-notifications-of-a-user

Reply:
Why is it not passed directly as an object?

Reply:
Because back then no one had a sample of what to do, so I just ported with the knowledge I had from follow vue docs and all the attributes of a notification are props:
https://github.com/nextcloud/notifications/blob/master/src/Components/Notification.vue#L77-L158
Made it explicit what is there and helps with type-ahead

Reply:
If props are already defined in the parent, I would just pass the data as a single object here then.